ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -1,56 +1,130 @@
 #include "MockProc.hpp"
+#include <map>
 #include <string>
 
-static std::string s_procName;
+using std::make_pair;
+using std::map;
+using std::string;
 
-int proc_pid(proc_t)
+static map<uintptr_t /*credential ID*/, int /* UID */> s_credentialMap;
+static map<vfs_context_t /*context*/, int /* UID */> s_contextMap;
+static map<int /*process Id*/, proc> s_processMap;
+static int s_selfPid;
+
+void MockProcess_Reset()
 {
-    return 1;
+    s_processMap.clear();
+    s_credentialMap.clear();
+    s_contextMap.clear();
 }
 
-void proc_name(int pid, char* buf, int size)
+void MockProcess_SetSelfPid(int selfPid)
+{
+    s_selfPid = selfPid;
+}
+
+int proc_pid(proc_t proc)
 {
-    strlcpy(buf, s_procName.c_str(), size);
+    return proc->pid;
 }
 
-void SetProcName(const std::string& procName)
+void proc_name(int pid, char* buf, int size)
 {
-    s_procName = procName;
+    map<int /*process Id*/, proc>::const_iterator found = s_processMap.find(pid);
+    if (found == s_processMap.end())
+    {
+        assert("pid not found in s_processMap");
```

Comment:
Out of curiosity, do these asserts do anything if they fire during a test?

Reply:
Doesn't look like it :-(, but we have a bunch of them in MockVnodeAndMount.  @pmj were you able to have them turned on during debugging?